### PR TITLE
doctor: add check/fix for bogus Action rows

### LINKED ISSUE #18650: 500 error accessing dashboard after upgrade to 1.16.1 from 1.15.10, caused by mismatched SQLite colmun type
### Gitea Version

1.16.1

### Git Version

2.25.1

### Operating System

Ubuntu Server 20.04 LTS

### How are you running Gitea?

Running as service with binaries (`gitea-1.16.1-linux-amd64`).

### Database

SQLite

### Can you reproduce the bug on the Gitea demo site?

No

### Description

After upgrading from `1.15.10` to `1.16.1` I get a 500 error when accessing the dashboard.
Other pages seem to work correctly if I open them directly (repos, issues, ...).

_(the same issue happened a week ago when I tried to upgrade from `1.15.10` to `1.16.0` but I haven't opened an issue at that time due to lack of time, sorry)_

Log contains only this relevant line:
```
.../web/feed/profile.go:22:RetrieveFeeds() [E] GetFeeds: Find: converting driver.Value type string to a int64: invalid syntax
```

Database version before update was `189`, after first run it was `209`.

I've reverted back to 1.15.10 (both binaries and DB).

### PR DESCRIPTION
**Pull request by singuliere**
See original [merge request from Loïc Dachary @ forgefriends](https://lab.forgefriends.org/forgefriends/forgefriends/-/merge_requests/49)

---

Related to https://github.com/go-gitea/gitea/pull/19598
Fixes https://github.com/go-gitea/gitea/issues/18650

### CODE REVIEW

File: modules/doctor/dbconsistency.go [hunk 1]
```diff
@@ -177,6 +177,9 @@ func checkDBConsistency(ctx context.Context, logger log.Logger, autofix bool) er
 		// find access without repository
 		genericOrphanCheck("Access entries without existing repository",
 			"access", "repository", "access.repo_id=repository.id"),
+		// find action without repository
```
Comment (lunny) [inline modules/doctor/dbconsistency.go:186]: Looks like it's a duplicated with line 177 - 179

Comment (singuliere) [reply modules/doctor/dbconsistency.go:186]: Line 177 is for **access** line 179 is for **action**

File: models/consistency.go [hunk 2]
```diff
@@ -246,3 +247,23 @@ func FixIssueLabelWithOutsideLabels() (int64, error) {
 
 	return res.RowsAffected()
 }
+
+// CountActionCreatedUnixString count created_unix that are a string
+func CountActionCreatedUnixString() (int64, error) {
```
Comment (Gusted) [inline models/consistency.go:252]: Does this count actions with a created_unix of a string type(everything is string in SQLite) or when it's empty. Seems like this comment isn't correct.

File: models/consistency_test.go [hunk 3]
```diff
@@ -103,3 +104,45 @@ func TestUpdateMilestoneCounters(t *testing.T) {
 	assert.NoError(t, issues_model.UpdateMilestoneCounters(db.DefaultContext, issue.MilestoneID))
 	unittest.CheckConsistencyFor(t, &issues_model.Milestone{})
 }
+
+func TestConsistencyUpdateAction(t *testing.T) {
+	assert.NoError(t, unittest.PrepareTestDatabase())
+	if setting.Database.UseSQLite3 {
```
Comment (Gusted) [inline models/consistency_test.go]: Maybe avoid indentation and do:
```go
if !setting.Database.UseSQLite3 {
	t.Skip("Test is only for SQLite database.")
}

File: models/consistency.go (hunk 2 above)
Comment (singuliere) [reply models/consistency.go:252]: only when it's an empty string, indeed

**Review by Gusted — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

File: models/consistency.go [hunk 4]
```diff
@@ -246,3 +247,23 @@ func FixIssueLabelWithOutsideLabels() (int64, error) {
 
 	return res.RowsAffected()
 }
+
+// CountActionCreatedUnixString count created_unix that are an empty string
```
Comment (delvh) [inline models/consistency.go]:
```suggestion
// CountActionCreatedUnixString count actions where created_unix is an empty string
```

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
Please update the branch

**Review by delvh — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Comment by 6543** [comment]
@singuliere can you "update branch" again :)

**Comment by 6543** [comment]
:rocket:

**Pull request merged**